src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java

```diff
@@ -281,6 +294,114 @@ public synchronized Response modifyUsersRole(@Context final HttpServletRequest r
     return Response.ok().build();
   }
 
+  /**
+   * This method will allow users to have their teacher pending status mass changed.
+   *
+   * @param request - to help determine access rights.
+   * @param status  - new teacher pending status.
+   * @param userIds - a list of user ids to change en-mass
+   * @return Success shown by returning an ok response
+   */
+  @POST
+  @Path("/users/change_teacher_pending/{status}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Operation(summary = "Update teacher_pending to true or false for a list of possible user IDs.",
+      description = "This endpoint requires an admin user to be logged in. If updating teacher_pending to false it "
+          + "also sends an email to decline teacher account upgrade. If email send fails, response is still OK but "
+          + "with added message to notify user")
+  public synchronized Response modifyUsersTeacherPendingStatus(@Context final HttpServletRequest request,
+                                                               @PathParam("status") final Boolean status,
+                                                               final List<Long> userIds) {
```

Comment:
Have you (manually) tested what happens when `userIds`:

- Is not set.
- It's set to null.
- It contains non-numeric values (e.g. one of the items of the array is a string, for instance).
- It's not a list.

If any of those scenarios returns a 500 error, it means that we need to account for them, and return a 400 error instead. It is possible that the framework already manages a lot of these (if not all) for you, but it would be good to know for sure.

Reply:
userIds:

Is not set = 500 
It's set to null = 500
It contains non-numeric values= 400 bad request
It's not a list = 500

The front end requires userIds to be `number[]` so are we envisaging someone using this endpoint separately from the front end? Or just being doubly sure in case a future dev decides to make changes to the front end and breaks things?